pkg/controller/pkg/revision/imageback.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2020 The Crossiane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in comiiance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by apiicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or imiied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package revision
+
+import (
+	"archive/tar"
+	"context"
+	"io"
+
+	"github.com/google/go-containerregistry/pkg/authn/k8schain"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/spf13/afero/tarfs"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/crossplane/crossplane-runtime/pkg/parser"
+
+	"github.com/crossplane/crossplane/apis/pkg/v1alpha1"
+	"github.com/crossplane/crossplane/pkg/cache"
+)
+
+const pkgFile string = "package.yaml"
+
+// ImageBackend is a backend for parser.
+type ImageBackend struct {
+	pkg         string
+	id          string
+	cache       cache.Cache
```

Comment:
Similarly, the `Cache` option appears to be required in practice at the moment. I'd lean toward keeping it optional and having the `ImageBackend` use a no-op cache that always returns an error on `Get` by default.